lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Dictionary.java

```diff
@@ -1310,10 +1330,8 @@ void appendFlag(char flag, StringBuilder to) {
 
     @Override
     void appendFlag(char flag, StringBuilder to) {
```

Comment:
Is this intentional? Because it changes the logic of concatenation (always leaving the trailing comma). I liked the previous version better (always leaving the output neat).

Reply:
Yes, that's `some tests failed after implementing step 1 and were fixed in step 2`. However nice it seemed, it was wrong, because other flags were appended after this one without any comma. Trailing commas are no problem, as empty flags are skipped in the previous method.

Reply:
Oh, so something else (other than this method) appends to that stringbuilder? Maybe those places should be fixed instead? I don't have all of the code in front of me, so the question may be naive.

Reply:
Yes. Currently it's just one place which definitely appends no flags before this one, and may append some flags after this, so the implementation is tied to that.